### PR TITLE
add support to setSchema to Config module

### PR DESCRIPTION
**Pull request by Madvinking**

### CODE REVIEW

**Comment by galangel** [comment]
remove .ds_store file

File: packages/config/src/Config.js [hunk 1]
```diff
@@ -10,7 +10,10 @@ export class Config {
   #observables;
 
   constructor(schema = null, config = {}) {
-    if (!schema || !Joi.isSchema(schema)) throw new Error('must pass Joi type schema');
+    if (schema && !Joi.isSchema(schema)) {
```
Comment (nofar9792) [inline packages/config/src/Config.js:13]: if ```schema``` is null, we won't throw an error. is it on purpose?

File: packages/config/src/Config.js [hunk 2]
```diff
@@ -38,6 +41,16 @@ export class Config {
     }
   }
 
+  setSchema(newSchema, config = this.#config) {
+    if (!Joi.isSchema(newSchema)) throw new Error('must pass Joi type schema');
```
Comment (nofar9792) [inline packages/config/src/Config.js:45]: same error?
create a function for joi validation

File: packages/config/src/Config.js (hunk 1 above)
Comment (nofar9792) [reply packages/config/src/Config.js:13]: ok i saw your test

File: packages/config/__test__/Config.test.js [hunk 3]
```diff
@@ -1,10 +1,8 @@
-import Joi from 'joi';
 import { Config } from '../src/Config.js';
+import Joi from 'joi';
 
 describe('Config - create config with schema', () => {
   it('should throw an error if no schema', () => {
```
Comment (sitbubu) [inline packages/config/__test__/Config.test.js:5]: should throw an error if not joi type schema

File: packages/config/__test__/Config.test.js [hunk 4]
```diff
@@ -77,6 +75,29 @@ describe('Config - create config with schema', () => {
     expect(() => new Config(schema, configs)).toThrow('"teams.name" must be a string');
   });
 
+  it('should throw error if setSchmea is not Joi', () => {
```
Comment (sitbubu) [inline packages/config/__test__/Config.test.js:78]: Should throw an error if on bad Joi validation(test until row 94 include)
Also from row 95 you can split to a successful test

**Comment by Madvinking** [comment]
@nofar9792 merge!

**Comment by nofar9792** [comment]
> @nofar9792 merge!

you're not my boss anymore
what don't you fix what we asked in the comments??
@AlonMiz @sitbubu  do you approve this PR?

**Comment by sitbubu** [comment]
> @nofar9792 merge!



> @nofar9792 merge!
No problem,
I'll merge it on Sunday, by then comments pls :)

**Comment by Madvinking** [comment]
> > @nofar9792 merge!
> 
> you're not my boss anymore
> why don't you fix what we asked in the comments??
> @AlonMiz @sitbubu do you approve this PR?
 @nofar9792  i did fix it

**Pull request merged**